ssz/sedes/serializable.py

```diff
@@ -132,24 +132,9 @@ def __len__(self):
         return len(self._meta.fields)
 
     def __eq__(self, other):
-        return self.__class__ is other.__class__ and hash(self) == hash(other)
+        return self.__class__ is other.__class__ and self.root == other.root
 
-    def __getstate__(self):
-        state = self.__dict__.copy()
-        # The hash() builtin is not stable across processes
-        # (https://docs.python.org/3/reference/datamodel.html#object.__hash__), so we do this here
-        # to ensure pickled instances don't carry the cached hash() as that may cause issues like
-        # https://github.com/ethereum/py-evm/issues/1318
-        state['_hash_cache'] = None
-        return state
-
-    _hash_cache = None
-
-    def __hash__(self):
```

Comment:
According to the [docs](https://docs.python.org/3/reference/datamodel.html#object.__hash__), two objects that are equal should have the same hash. So we should probably not remove it, but replace it with something like

```Py
return hash(self.__class__) * int.from_bytes(self.root, "little")
```